Patch-mainline: never
From: <ohering@suse.de>
Subject: hyperv: force vmbus interrupts to cpu 0

For the time being bind channels to irq 0, otherwise vmbus will fail with more than one vcpu.

---
 drivers/hv/channel_mgmt.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: linux-3.8-rc3-master/drivers/hv/channel_mgmt.c
===================================================================
--- linux-3.8-rc3-master.orig/drivers/hv/channel_mgmt.c
+++ linux-3.8-rc3-master/drivers/hv/channel_mgmt.c
@@ -298,6 +298,7 @@ static const uuid_le hp_devs[] = {
 };
 
 
+#if 0
 /*
  * We use this state to statically distribute the channel interrupt load.
  */
@@ -339,6 +340,10 @@ static u32 get_vp_index(uuid_le *type_gu
 	cur_cpu = (++next_vp % max_cpus);
 	return hv_context.vp_index[cur_cpu];
 }
+#else
+/* Legacy IRQs can not be handled on all cpus, bind to cpu 0. */
+static inline u32 get_vp_index(uuid_le *type_guid) { return 0; }
+#endif
 
 /*
  * vmbus_onoffer - Handler for channel offers from vmbus in parent partition.
